object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/AutoValueSupport.java

```diff
@@ -0,0 +1,388 @@
+package org.checkerframework.checker.objectconstruction.framework;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import com.sun.source.tree.NewClassTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Types;
+import com.sun.tools.javac.processing.JavacProcessingEnvironment;
+import java.beans.Introspector;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.objectconstruction.ObjectConstructionAnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.util.AnnotatedTypes;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/**
+ * AutoValue Support for object construction checker, add CalledMethods annotation to the code
+ * generated by AutoValue
+ */
+public class AutoValueSupport implements FrameworkSupport {
+
+  private ObjectConstructionAnnotatedTypeFactory atypeFactory;
+
+  public AutoValueSupport(ObjectConstructionAnnotatedTypeFactory atypeFactory) {
+    this.atypeFactory = atypeFactory;
+  }
+
+  /**
+   * add CalledMethods annotation to a constructor call inside a generated toBuilder implementation
+   * in AutoValue
+   *
+   * @param tree
+   * @param type
```

Comment:
If you're going to add `@param` to the Javadoc, you should add some information there about what that parameter does. If there's nothing to say, don't include `@param`.